### PR TITLE
Adjust template for solvers

### PR DESCRIPTION
**Pull request by pacospace**
Depends-On: https://github.com/thoth-station/common/pull/705

Signed-off-by: Francesco Murdaca <fmurdaca@redhat.com>

### CODE REVIEW

**Comment by pacospace** [comment]
recheck

File: openshift/solve_and_sync-template.yaml [hunk 1]
```diff
@@ -118,10 +158,217 @@ objects:
...
+        operator: graph-sync
+        task: solver
+        mark: cleanup
+    spec:
+      template:
+        metadata:
+          labels:
+            app: thoth
+            component: solver
+            solver-type: solver-rhel-8-py36
```
Comment (fridex) [inline openshift/solve_and_sync-template.yaml]: Can this become a parameter? That way we could maintain just one template for all the solver types.

File: workflows/templates/solver-template.yaml [hunk 2]
```diff
@@ -73,8 +74,8 @@ spec:
             mountPath: /mnt/workdir
         resources:
           limits:
-            cpu: 500m
-            memory: 512Mi
+            cpu: 250m
+            memory: 256Mi
```
[the remark this answers is not in the document]
Comment (pacospace) [reply workflows/templates/solver-template.yaml]: There are solver failing already for OOM, so you are right :) I will increase it again

File: openshift/solve_and_sync-template.yaml (hunk 1 above)
Comment (pacospace) [reply openshift/solve_and_sync-template.yaml]: not really due to the `get_solver_names`(https://github.com/thoth-station/common/blob/5cea9e9f09457d64047d381031e79b54741e364c/thoth/common/openshift.py#L690) method used at the moment to check for solver names by different services (e.g. graph-refresh-job, metrics-exporter). We should think about an alternative solution.

Comment (fridex) [reply openshift/solve_and_sync-template.yaml]: Right :+1: Hm... what about putting solvers into thoth's configmap? IMHO it worth to maintain just one solver workflow template.

Comment (pacospace) [reply openshift/solve_and_sync-template.yaml]: Yes solvers in Thoth CM would be a solution, there would be key, value for each solver present right?

Comment (fridex) [reply openshift/solve_and_sync-template.yaml]: What about an entry like this (as CM do not support nested configuration):

```
solvers: |
   solver1
   solver2
```

The code in thoth-common would load the configmap and check for "solvers" key, subsequently parse solver names out of it:

```python
cm = _OPENSHIFT.get_configmap("thoth", namespace="thoth-infra-stage")
solvers = [s.strip() for s in cm["solvers"].splitlines()]
```

Comment (pacospace) [reply openshift/solve_and_sync-template.yaml]: Yes! https://github.com/thoth-station/common/pull/715

**Comment by pacospace** [comment]
recheck

**Comment by harshad16** [comment]
recheck

**Comment by harshad16** [comment]
recheck

**Comment by pacospace** [comment]
recheck

**Comment by harshad16** [comment]
recheck

**Pull request merged**